students/JasneetChandok/lesson02/assignment/src/charges_calc.py

```diff
@@ -5,43 +5,102 @@
+    console_handler.setFormatter(formatter)
+
+    logger = logging.getLogger()
+
+    if level == '2':
+        file_handler.setLevel(logging.ERROR)
+        console_handler.setLevel(logging.ERROR)
+        logger.setLevel(logging.ERROR)
+    elif level == '3':
+        file_handler.setLevel(logging.WARNING)
+        console_handler.setLevel(logging.WARNING)
+        logger.setLevel(logging.WARNING)
+    elif level == '4':
+        file_handler.setLevel(logging.WARNING)
+        console_handler.setLevel(logging.DEBUG)
+        logger.setLevel(logging.DEBUG)
+    else:  
+        file_handler.setLevel(logging.CRITICAL)
+        console_handler.setLevel(logging.CRITICAL)
+        logger.setLevel(logging.CRITICAL)
+
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+
+    return logger
+
+
 def load_rentals_file(filename):
     with open(filename) as file:
         try:
             data = json.load(file)
-        except:
+            logger.debug(f'Successfully loaded {len(data.keys())} elements from {filename}')
+        except Exception as e:
+            logger.error(f'Could not import date from {filename}: {e}')
             exit(0)
     return data
 
+
 def calculate_additional_fields(data):
     for value in data.values():
-        try:
-            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
-            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
-            value['total_days'] = (rental_end - rental_start).days
-            value['total_price'] = value['total_days'] * value['price_per_day']
-            value['sqrt_total_price'] = math.sqrt(value['total_price'])
-            value['unit_cost'] = value['total_price'] / value['units_rented']
-        except:
-            exit(0)
+        if value['rental_start'] > value['rental_end']:
+            logger.error('start date before end date. Skipping...')
+        else:
+            try:
+                rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+                logger.debug(f'calculated rental_start: {rental_start}')
+                rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+                logger.debug(f'calculated rental_end: {rental_end}')
+                value['total_days'] = (rental_end - rental_start).days
+                logger.debug(f'calculated total_days: {value["total_days"]}')
+                value['total_price'] = value['total_days'] * value['price_per_day']
+                logger.debug(f'calculated total_price: {value["total_price"]}')
+                value['sqrt_total_price'] = math.sqrt(value['total_price'])
+                logger.debug(f'calculated sqrt_total_price: {value["sqrt_total_price"]}')
+                value['unit_cost'] = value['total_price'] / value['units_rented']
+                logger.debug(f'calculated unit_cost: {value["unit_cost"]}')
+            except Exception as e:
```

Comment:
Be careful here, you don't want to catch the general Exception.  I would recommend being a little careful when catch exceptions because when you expect errors, you should generally know what they are so that you know you are handling them appropriately. For example, you would handle a DivideByZero exception differently than a FileNotFound. The 'Exception' will catch both but you may have different behaviors on what to do after finding the error.